### PR TITLE
[Mass+] Fix search by tag

### PR DESCRIPTION
**Pull request by AprilSylph**
now that the Mass Post Editor can also work with your drafts and queue, assuming the blogname comes after `/mega-editor/` in the URL is incorrect - changed this logic to pluck out the last part of the pathname every time

### CODE REVIEW

**Comment by cyle** [comment]
> "/mega-editor/published/cyle".split("/").slice(-1)[0]
> "cyle"

👍 

but this doesn't work if you have a date selected 😭 

> "/mega-editor/published/cyle/2019/2".split("/").slice(-1)[0]
> "2"

👎 

i think this'll work in all scenarios:

```js
var blog_shortname = location.pathname.split("/")[3];
```

**Comment by AprilSylph** [comment]
ooh i didnt notice date selection. thank you!

File: Extensions/mass_plus.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ XKit.extensions.mass_plus = new Object({
 		if (last_timestamp.indexOf(" ") !== -1) {
 			last_timestamp = last_timestamp.substring(0, last_timestamp.indexOf(" "));
 		}
-		var blog_shortname = location.pathname.split("/").slice(-1)[0];
+		var blog_shortname = location.pathname.split("/")[3];
 		if (blog_shortname .indexOf("/") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("/")); }
```
Comment (nightpool) [inline Extensions/mass_plus.js]: not sure how this code could ever be reached

File: Extensions/mass_plus.js [hunk 2]
```diff
@@ -171,7 +171,7 @@ XKit.extensions.mass_plus = new Object({
...
 		if (blog_shortname .indexOf("?") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("?")); }
 		if (blog_shortname .indexOf("#") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("#")); }
```
Comment (nightpool) [inline Extensions/mass_plus.js]: not sure how any of this code could be reached actually, `location.pathname` doesn't include hash locators or query params, they're in `.hash` and `.search` respectively.

File: Extensions/mass_plus.js [hunk 3]
```diff
@@ -171,7 +171,7 @@ XKit.extensions.mass_plus = new Object({
 		if (last_timestamp.indexOf(" ") !== -1) {
 			last_timestamp = last_timestamp.substring(0, last_timestamp.indexOf(" "));
 		}
-		var blog_shortname = location.pathname.split("/").slice(-1)[0];
+		var blog_shortname = location.pathname.split("/")[3];
```
Comment (nightpool) [inline Extensions/mass_plus.js]: might be clearer as array destructuring indicating what you expect the other components of the url to be? Something like:

`var [_root, _page, _post_type, blog_shortname] = location.pathname.split("/")`

Comment (AprilSylph) [reply Extensions/mass_plus.js]: had to not actually assign variables because of `no-unused-vars` but they're labelled now!

**Pull request merged**